onnxruntime/contrib_ops/cpu/quantization/dynamic_quantize_matmul.cc

```diff
@@ -161,6 +161,115 @@ class DynamicQuantizeMatMul final : public MatMulIntegerToFloatBase {
 
   Status Compute(OpKernelContext* context) const override;
 
+  Status PrePack(const Tensor& tensor, int input_idx, AllocatorPtr alloc,
+                 /*out*/ bool& is_packed,
+                 /*out*/ PrePackedWeights* prepacked_weights) override {
+    // only pack Matrix B
+    if (input_idx == GetBIdx()) {
+      const Tensor* b_zp_tensor{nullptr};
+
+      // assume zero is present
+      auto b_zp_present{true};
+
+      // zero point tensor could be provided as a direct input to the kernel and not as a constant so this
+      // test is not sufficient
+      const OrtValue* b_zp;
+      if (Info().TryGetConstantInput(IN_B_ZERO_POINT, &b_zp)) {
+        b_zp_tensor = &b_zp->Get<Tensor>();
+      }
+
+      // MlasDynamicQgemm requires symmetric quantization for B, so no zero point should exist or it should
+      // have a zero value
+      if (b_zp_tensor != nullptr) {
+        b_zp_present = (b_zp_tensor->Shape().Size() != 1) ||
+                       (static_cast<const uint8_t*>(b_zp_tensor->DataRaw())[0] != 0);
+      }
+
+      // MlasDynamicQgemm requires scale data to be available at packing stage
+      const OrtValue* b_scale;
+      auto b_scale_available = Info().TryGetConstantInput(IN_B_SCALE, &b_scale);
+
+      CanUseDynamicQuantMLAS = (!b_zp_present && b_scale_available);
```

Comment:
Any checks around broadcasting (i..e) multi-dimensional weights/activations to be included here ?